Conda

paskino opened the pull request:
initial working recipe for linux.

OSX would need tweak on build.sh. 

The branch should be aligned with setup_py #109

paskino committed:
first working conda recipe

paskino committed:
first working conda recipe

paskino committed:
use PYTHON_DEST to point to the site_packages

paskino committed:
removed SIRF_VERSION env var

paskino committed:
use PYTHON_DEST to point to the site_packages

paskino committed:
do not overwrite the PYTHON_DEST?

paskino committed:
removed messages

paskino committed:
updated about, removed SIRF_VERSION env var

paskino committed:
updated about, removed SIRF_VERSION env var

paskino committed:
updated about, removed SIRF_VERSION env var

paskino committed:
pulls the 1.0.0 tag

paskino committed:
removed SIRF_VERSION env var

paskino committed:
removed SIRF_VERSION env var

casperdcl committed:
initial setup.py

- depends on https://github.com/CCPPETMR/SIRF/pull/161
- fixes (partially) #107
- TODO: alias p(Gadgetron|STIR|Utilities) -> sirf.p*
- perhaps use `setup.py install` not `pip install`?
  - or drop `-e` argument in `pip install`?

casperdcl committed:
backward-compatible setup.py

- depends on https://github.com/CCPPETMR/SIRF/pull/160
- fixes #107

paskino committed:
Merge remote-tracking branch 'origin/setup_py' into conda

casperdcl committed:
missing packages

paskino committed:
added missing package

paskino committed:
bugfixes

paskino committed:
bugfixes

paskino committed:
bugfixes

paskino committed:
Merge remote-tracking branch 'origin/setup_py' into conda

paskino committed:
bugfixes

paskino committed:
bugfixes

casperdcl committed:
pyvenv -> miniconda

paskino committed:
copies the .so into the site-packages

paskino committed:
copies the .so into the site-packages

casperdcl committed:
pip -> setup.py

casperdcl committed:
missing docker devel cache

paskino committed:
added dependency

paskino committed:
added dependency

paskino committed:
added dependency

paskino committed:
Merge remote-tracking branch 'origin/setup_py' into conda

casperdcl commented:
1. merge https://github.com/CCPPETMR/SIRF/pull/160
2. merge #109
3. rebase this and merge

close #106?

paskino commented:
I agree with you.

paskino committed:
removed boost dependency

currently the boost in conda-forge does not contain all the libs we need. So we need to build and install.

paskino committed:
removed boost dependency

currently the boost in conda-forge does not contain all the libs we need. So we need to build and install.

paskino committed:
removed boost dependency

currently the boost in conda-forge does not contain all the libs we need. So we need to build and install.

KrisThielemans commented:
should we add a conda build to Travis? Maybe it's already done via the docker build.

that'd allow you to test on OSX.

paskino committed:
Added dependency

paskino committed:
Added dependency

paskino committed:
Added dependency

paskino committed:
build latest SIRF, use pre-built boost

casperdcl committed:
update SIRF tag to current master

paskino committed:
build Boost, remove dependency on Boost in meta.yaml

KrisThielemans commented:
Data and examples is #102.

evgueni-ovtchinnikov commented:
@casperdcl: so far as i can see examples in MR/interactive require simulated_MR_2D_cartesian.h5, which is not in https://github.com/CCPPETMR/SIRF_data/tree/master/examples

@ckolbPTB: why did you not use existing_filepath function? the error message would be more meaningful

also, the graphics in coil_combination.py crashes python when run from command line (spyder survives)

KrisThielemans commented on conda-recipe/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+    - h5py
+    - libxml2
+    - psutil
+    - nose
```
`nose` and `coverage` not needed for conda package? (they probably don't make sense)

paskino replied:
They are needed during the build

KrisThielemans replied:
not convinced. they are needed for the testing. but are you testing here?
not a problem to leave them in if it helps of course.

paskino replied:
It wouldn't build otherwise

KrisThielemans replied:
well... in that case, we need it! :-)

KrisThielemans commented on conda-recipe/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+    - scipy
+    - matplotlib
+    - docopt
+    - h5py
```
99% sure we don't need this

KrisThielemans commented on conda-recipe/build.sh:
```diff
@@ -0,0 +1,69 @@
+
+cd $SRC_DIR/build
+
+echo "$SRC_DIR/ccpi/Python"
```
really?

paskino replied:
Really what?

KrisThielemans replied:
ccpi...

paskino replied:
You know, I'm torn

KrisThielemans commented on conda-recipe/build.sh:
```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+#
+# Script to install/update the CCP-PETMR VM. It could also be used for any other system
+# but will currently change your .sirfrc. This is to be avoided later on.
```
incorrect comment

KrisThielemans commented on conda-recipe/build.sh:
```diff
@@ -0,0 +1,69 @@
+#if [ -z "$SIRF_VERSION" ]; then
+#    echo "Need to set SIRF_VERSION"
+#    exit 1
+#fi  
```
until we fix this, will need to add it to the checklist

KrisThielemans commented on conda-recipe/meta.yaml:
```diff
@@ -0,0 +1,57 @@
+    - fftw
+    - libgcc-ng
+    - libstdcxx-ng
+    - openblas
```
not for build?

casperdcl replied:
I feel like there's a difference between build, install (runtime), and test. I'd think nose and coverage fall into the last category, and things like h5py for install? Is there a way to specify these three categories for conda? I'm sure at least two are possible. At least split into sections with comments

paskino replied:
Indeed there is. Currently The tricky bit is afaik that the install phase of the external packages is run during the make phase.

KrisThielemans reviewed (changes requested):
could probably easily split ismrmrd and gadgetron off...

paskino commented:
I guess we'll end up creating several recipes which create the packages sirf depends on, and sirf.
As we discussed, we could pass a parameter to the superbuild and build only part of it. Now I don't remember which parameter is exactly.

KrisThielemans commented:
currently you'd run CMake as usual. then just say `make gadgetron` or whatever. A bit wasteful, but minimal effort...

KrisThielemans commented:
this is fine-tuning of course. leave it for later!

paskino commented:
I wouldn't mind issuing cake 3 or 4 times!

paskino committed:
Merge branch 'conda' of https://github.com/CCPPETMR/SIRF-SuperBuild into conda

Added a bunch of conda recipes to build:
1. boost
2. gtest
3. ismrmrd
4. gadgetron
5. armadillo
6. stir (incomplete)

fixes #126

paskino committed:
Merge branch 'conda' of https://github.com/CCPPETMR/SIRF-SuperBuild into conda

paskino committed:
first working compilation of SIRF with conda

casperdcl commented:
> I guess we'll end up creating several recipes which create the packages sirf depends on, and sirf.

helps that STIR is now in `conda-forge`

KrisThielemans commented:
> helps that STIR is now in conda-forge

First need to resolve https://github.com/UCL/STIR/issues/211

paskino committed:
Merge remote-tracking branch 'origin' into conda

paskino commented:
Directories to pass to `External_SIRF.cmake` are defined 

* For STIR [here](https://github.com/CCPPETMR/SIRF-SuperBuild/blob/master/SuperBuild/External_STIR.cmake#L128L130)
* NIFTYREG [here](https://github.com/CCPPETMR/SIRF-SuperBuild/blob/master/SuperBuild/External_NIFTYREG.cmake#L72L74)
* ITK (needed by [STIR](https://github.com/CCPPETMR/SIRF-SuperBuild/blob/master/SuperBuild/External_STIR.cmake#L109)) where?

paskino committed:
fixed dependency

paskino committed:
changed SIRF version; add ITK

paskino committed:
allows build of Zlib if not present on system

paskino committed:
updated gadgetron version

paskino committed:
added build variants file

paskino committed:
added hdf5

paskino committed:
additional fixes

paskino committed:
updated recipe

paskino committed:
added build variants for armadillo

paskino committed:
Merge tag 'v2.1.0' into conda

version 2.1.0

paskino committed:
require Boost 1.68

paskino committed:
install in CONDA build environment

paskino committed:
fix run boost requirement

KrisThielemans commented on conda/boost/build.sh:
```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
```
I would try to avoid building boost at all costs. Use the conda dependency

KrisThielemans commented on conda/hdf5/build.sh:
```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
```
let's not build hdf5

KrisThielemans commented on conda/ismrmrd/build.sh:
```diff
@@ -0,0 +1,72 @@
+    -DPYTHON_DEST_DIR=${PREFIX}/python\
+    -USIRF_URL \
+    -USIRF_TAG \
+    -DSIRF_TAG=v2.1.0\
```
let's not put explicit versions in any of the conda build.sh. In any case, SIRF version is irrelevant in ismrmrd/build.sh

KrisThielemans commented on conda/sirf/build.sh:
```diff
@@ -0,0 +1,82 @@
+
+make -j2 ITK
+make -j2 NIFTYREG
+make -j2 STIR
```
no need for these.

KrisThielemans commented on conda/stir/build.sh:
```diff
@@ -0,0 +1,76 @@
+    -DUSE_SYSTEM_GTest=On\
+    -DCONDA_BUILD=On
+
+make  -j1 Gadgetron
```
in stir/build.sh?

KrisThielemans reviewed (changes requested):
I haven't checked this really, but I'm confused about the various `build.sh` files, and then `sirf/build.sh` building other stuff anyway.

paskino committed:
updated version strings

paskino committed:
added niftyreg

paskino committed:
updated meta

KrisThielemans commented:
STIR 4.0.1 conda might work ok. There's https://anaconda.org/inati/ismrmrd but it's out-of-date. Might be worth checking with them.

KrisThielemans commented:
By the way, I'm not 100% sure what the strategy here was/is, but I think we should have a SIRF-only conda package (with its dependencies, most obtained via conda). I would at least initially not attempt gadgetron-via-conda.

paskino committed:
Merge remote-tracking branch 'origin' into conda

paskino committed:
work in progress

paskino committed:
added missing_dso_whitelist

paskino committed:
add _openmp_mutex

paskino committed:
updated recipes

KrisThielemans commented:
See https://github.com/gadgetron/gadgetron/pull/1036 for Gadgetron's `conda` efforts

KrisThielemans commented on conda-recipe/meta.yaml:
```diff
@@ -0,0 +1,60 @@
+    - vc 14 # [win and py35]
+    - vc 9  # [win and py27]
+    - python 3.6 # [py36]
+    - python 2.7 # [py27]
```
drop this of course
